modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateTimeInput.es.js

```diff
@@ -43,65 +47,87 @@ class DateTimeInput extends React.Component {
 		return returnVal;
 	}
 
-	_handleDateChange = (event) => {
-		const value = event.target.value;
-
+	_handleDateChange = (value) => {
 		this.setState({value});
 	};
 
-	_handleDateBlur = (event) => {
-		const date = dateFns.format(event.target.value, INPUT_DATE_FORMAT);
-
-		if (date !== 'Invalid Date') {
-			this.setState(
-				{
-					value: date,
-				},
-				() => {
-					this.props.onChange({
-						type: PROPERTY_TYPES.DATE_TIME,
-						value: dateFns
-							.parse(date, INPUT_DATE_FORMAT)
-							.toISOString(),
-					});
-				}
-			);
+	_handleExpandedChange = (expandedState) => {
+		this.setState({expanded: expandedState});
+
+		if (expandedState === false) {
+			this._saveDateTimeValue();
 		}
-		else {
-			const resetDate = dateFns.format(new Date(), INPUT_DATE_FORMAT);
-
-			this.setState(
-				{
-					value: resetDate,
-				},
-				() => {
-					this.props.onChange({
-						type: PROPERTY_TYPES.DATE_TIME,
-						value: dateFns
-							.parse(resetDate, INPUT_DATE_FORMAT)
-							.toISOString(),
-					});
-				}
-			);
+	};
+
+	_saveDateTimeValue = () => {
+		const date = dateFns.format(this.state.value, INPUT_DATE_FORMAT);
+
+		if (this.previousValue !== date) {
+			if (date !== 'Invalid Date') {
+				this.setState(
+					{
+						value: date,
+					},
+					() => {
+						this.props.onChange({
+							type: PROPERTY_TYPES.DATE_TIME,
+							value: dateFns
+								.parse(date, INPUT_DATE_FORMAT)
+								.toISOString(),
+						});
+					}
+				);
+			}
+			else {
+				const resetDate = dateFns.format(new Date(), INPUT_DATE_FORMAT);
+
+				this.setState(
+					{
+						value: resetDate,
+					},
+					() => {
+						this.props.onChange({
+							type: PROPERTY_TYPES.DATE_TIME,
+							value: dateFns
+								.parse(resetDate, INPUT_DATE_FORMAT)
+								.toISOString(),
+						});
+					}
+				);
+			}
 		}
+		this.previousValue = date;
 	};
 
 	render() {
-		const {value} = this.state;
+		const {expanded, value} = this.state;
 		const {disabled, propertyLabel} = this.props;
 
 		return (
 			<div className="criterion-input date-input">
-				<input
+				<ClayDatePicker
 					aria-label={`${propertyLabel}: ${Liferay.Language.get(
 						'select-date'
 					)}`}
-					className="form-control"
-					data-testid="date-input"
```

Comment:
If we delete the data-testid the QA tests will break, right? 

@CarlosBrichete

Reply:
I'm not really sure, that property is used in the front-end tests